src/PrestaShopBundle/ApiPlatform/Resources/Hook.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\ApiPlatform\Resources;
+
+use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\Put;
+use PrestaShop\PrestaShop\Core\Domain\Hook\Command\UpdateHookStatusCommand;
+use PrestaShop\PrestaShop\Core\Domain\Hook\Exception\HookNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Hook\Query\GetHookStatus;
+use PrestaShop\PrestaShop\Core\Domain\Hook\QueryResult\HookStatus;
+use PrestaShopBundle\ApiPlatform\Processor\CommandProcessor;
+use PrestaShopBundle\ApiPlatform\Provider\QueryProvider;
+
+#[ApiResource(
+    operations: [
+        new Get(
+            uriTemplate: '/hook-status/{id}',
+            requirements: ['id' => '\d+'],
+            exceptionToStatus: [HookNotFoundException::class => 404],
+            output: HookStatus::class,
+            provider: QueryProvider::class,
+            extraProperties: ['query' => GetHookStatus::class]
+        ),
+        new Put(
+            uriTemplate: '/hook-status',
+            processor: CommandProcessor::class,
+            extraProperties: ['command' => UpdateHookStatusCommand::class]
+        ),
+    ]
+)]
+class Hook
+{
+    /**
+     * @var int
+     */
+    public int $id = 0;
```

Comment:
Do we really need this field since we already have `$hookId` Is this needed only to match the API placeholder `{id}` If so I wonder if there's a possibility to use `{id}` in the URI but indicate some mapping to make it match with `hookId` ?

If not then maybe we should simply use `hookId` as the placeholder, but globally I'd prefer we don't add addition/duplicate fields like this one